### DISCUSSION TITLE
[Feature Request] Combined Trigger/Measured flag

### DISCUSSION THREAD

**Discussion by suspect15** [RetroAchievements/RAIntegration#1043] [Ideas]
Currently the only way to have the same hits target or value with both a Trigger and Measured flag is to utilize Alt groups since the challenge indicator will not display until the Measured condition(s) are true, which generally is when the achievement would award.  This isn't a significant burden, however it would help streamline a not insignificant number of achievements if there was a way to flag things as both Trigger and Measured simultaneously in the same group and have both effects be active.

**Comment by redwizard42** [comment]
This is probably the better place to ask for it hah.

Right now when I need both I usually have a long chain of things added together in one alt with Measured, and the same chain in another alt with Trigger.  Having a combo flag would have a big halving effect (as far as number of conditions go) on that kind of logic, with is a fairly typical use case of when you need both.

**Comment by Jamiras** [comment]
If you're already splitting the logic into a Measured Alt and a Trigger Alt, you can simplify the Trigger Alt to just be Trigger(false). Here's an example I made for SMB:
![image](https://github.com/RetroAchievements/RAIntegration/assets/32680403/eeac5d9e-462d-459e-84ad-061f00cf296c)

If everything in Core is true, then the Trigger icon will show because Core+Alt2 would trigger if the Trigger condition were true. If anything in Core becomes false, the Trigger icon would disappear.

As soon as the Measured target is reached, Core+Alt1 would cause the achievement to trigger.

**Reply by suspect15** [reply]
This is a really good example.  I was able to significantly reduce the size of an achievement using it from 23 conditions to 11.  It's definitely tough to argue with results, but I do think there's value in a combined flag from at least a future maintenance perspective as it would likely be easier for someone to understand the logic more quickly if it was contained in a single group, but I do acknowledge that the value is pure convenience, not necessity.  Nevertheless - I definitely appreciate the example as it was a great lesson for me.

**Reply by redwizard42** [reply]
I agree this works, but it's inherently a kludge and makes reading logic harder (as did the less efficient twin alts).  Someone will stumble over this oddity when fixing a ticket.
